src/NuGet.Core/NuGet.PackageManagement/IDE/PackageRestoreManager.cs

```diff
@@ -69,15 +72,42 @@ public virtual async Task RaisePackagesMissingEventForSolutionAsync(string solut
             PackagesMissingStatusChanged?.Invoke(this, new PackagesMissingStatusEventArgs(missing));
         }
 
-        public virtual void RaiseAssetsFileMissingEventForProjectAsync(bool isAssetsFileMissing)
+        /// <summary>
+        /// Checks if the project is missing an assets file and updates the visibility
+        /// </summary>
+        public virtual async Task RaiseAssetsFileMissingEventForProjectAsync()
         {
             if (_assetsFileMissingStatusChanged != null)
             {
                 foreach (var handler in _assetsFileMissingStatusChanged.GetInvocationList())
                 {
                     try
                     {
-                        handler.DynamicInvoke(this, isAssetsFileMissing);
+                        var nuGetProject = await SolutionManager.GetNuGetProjectAsync(_packageRestoreBarProjectId);
```

Comment:
I'm not sure I understand why we need all this refactoring. 
This check can be performed in the listener no? In particular the PackageRestoreBar?